tests/trainer/test_dataloaders.py

```diff
@@ -1470,35 +1471,28 @@ def __iter__(self):
         def __next__(self):
             return next(self._iter)
 
-    class DataLoaderFunc:
-        def __init__(self, loader):
-            self.loader = loader
-
-        def __call__(self):
-            return self.loader
-
     class TestModel(BoringModel):
         def __init__(self):
             super().__init__()
-            self.on_train_dataloader_called = False
+            self.train_dataloader_called = False
             self.on_train_batch_start_called = False
-            self.on_val_dataloader_called = False
+            self.val_dataloader_called = False
             self.on_val_batch_start_called = False
 
-        def on_train_dataloader(self) -> None:
-            loader = self.train_dataloader()
-            self.train_dataloader = DataLoaderFunc(DataLoaderWrapper(loader))
-            self.on_train_dataloader_called = True
+        def train_dataloader(self) -> TRAIN_DATALOADERS:
```

Comment:
let's not test these again, this is duplicated since we already have tests that these hooks get called.
and besides, counting how many times it is called is more important to know vs. if they are called, so I would even remove the whole test here.